codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/BaseClientBuilderClass.java

```diff
@@ -406,8 +406,10 @@ private MethodSpec finalizeServiceConfigurationMethod() {
         builder.addStatement("builder.option($1T.EXECUTION_INTERCEPTORS, interceptors)", SdkClientOption.class);
 
         if (model.getCustomizationConfig().getServiceConfig().hasDualstackProperty()) {
+            builder.beginControlFlow("if (dualStackConfiguredExplicitly)");
             builder.addStatement("builder.option($T.DUALSTACK_ENDPOINT_ENABLED, finalServiceConfig.dualstackEnabled())",
                             AwsClientOption.class);
```

Comment:
Would `serviceConfigBuilder.dualstackEnabled()` instead of `finalServiceConfig.dualstackEnabled()` work, without needing to mess with the `dualStackConfiguredExplicitly` flag?

Reply:
Yeah this should work. Great suggestion!